test/parallel/test-worker-heap-snapshot.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+// Flags: --expose-internals
+const common = require('../common');
+const assert = require('assert');
+const { Worker } = require('worker_threads');
+const { HeapSnapshotStream } = require('internal/heap_utils');
+const { kHandle } = require('internal/worker');
+
+const worker = new Worker(__filename);
+const snapShotResult = { ondone: () => {} };
+worker[kHandle].takeHeapSnapshot = common.mustCall(() => snapShotResult);
+
+worker.on('online', () => {
```

Comment:
We need to ensure that the `assert.ok` will get called here:
```suggestion
worker.on('online', mustCall(() => {
```

Reply:
omg! I thought I did it hahaha. Just fixed! thank you!